src/main/java/org/candlepin/model/EnvironmentCurator.java

```diff
@@ -141,34 +139,14 @@ public List<Environment> listWithContent(Set<String> contentIds) {
      *  the number of environments deleted
      */
     public int deleteEnvironmentsForOwner(Owner owner) {
-        String jpql = "SELECT e.id FROM Environment e WHERE e.owner.id = :owner_id";
+        String jpql = "DELETE FROM Environment env WHERE env.owner.id = :owner_id";
```

Comment:
Hibernate is taking care of deleting the environment-content data here right?

Reply:
It should be, yes. If it weren't, we'd get angry FK violations. I suspect it might be limited to what we specify should cascade, but in the case of environment-content, we're cascading everything.